### DISCUSSION TITLE
希望能出一期自定义控件的教程

### DISCUSSION THREAD

**Discussion by Gaoyifei1011** [BYJRK/DotNet-Discussions#6] [通用]
1.之前在使用WinUI Library的ComboBox，发现当ComboBox下拉框打开后，会发现编辑框（无编辑模式）会被下拉框挡住，和WPF的ComboBox的下拉框格式不太相同。如果可以的话，能否使用DropDownButton和MenuFlyout（RadioButton）组合实现一个不需要编辑模式的下拉框控件呢？
默认的下拉框控件 
![DefaultComboBox](https://user-images.githubusercontent.com/49179966/211303989-58b1ef8a-90cd-403a-81c3-f1162d30f631.png)
希望能展示出来的下拉框控件
![DeramfulComboBox](https://user-images.githubusercontent.com/49179966/211303992-5e124b80-d5d1-4781-a491-5da1534ff149.png)

**Comment by Gaoyifei1011** [comment]
好像我见只有一个WPF的选项，没有UWP/Windows App SDK选项，希望能增加一个

**Comment by BYJRK** [comment]
UWP以来的ComboBox就是这样的一种设计语言，用一个类似popup的东西覆盖在box上方供用户选择，这样可以减少用户在打开“下拉菜单”之后鼠标（或手指）需要额外移动的距离。我觉得这样也挺好的。你有什么必要让它和WPF的传统方式保持一致吗？

**Reply by Gaoyifei1011** [reply]
您好，感谢您的回复。

关于这样做的设计目的，其实坦白说，并没有太多的个人想法，只是单纯觉得不好看而已。😂😂😂 

之前我在Microsoft Store下载过一个名为WPF UI框架，还有体验过Avalonia的框架，发现他们的下拉框都是这样设计的，感觉这样设计既保留了旧版的样式，同时还有了Win11的设计。
![image](https://user-images.githubusercontent.com/49179966/211449702-f3bdf9d0-ec6b-4669-b058-3832e6fada0e.png)

这种想法是一次我的电脑上面自带的UWP应用照片升级到最新版的时候发现的。打开新版照片的设置发现就是这样设计的（虽然是手动添加设置选项） 
![image](https://user-images.githubusercontent.com/49179966/211449426-83343eb7-12a0-4cb6-a701-8bdba1d9b8a7.png)
正好我在最近一段时期也在学习C#，所以有机会的话可以通过学习自定义控件的方式实现一个相对来说更符合这种设计的下拉框

**Comment by eijjxx** [comment]
如果学做自定义控件的话，建议看一下handycontrol的源码
我感觉比material design in xaml简单一些
基本上看懂了handycontrol，大部分的自定义控件就能做了

**Reply by Gaoyifei1011** [reply]
好的，非常感谢
